src/main/java/pl/nogacz/snake/board/Board.java

```diff
@@ -94,14 +98,76 @@ private void moveSnakeHead(Coordinates coordinates) {
                     tailLength++;
 
                     snakeHeadCoordinates = coordinates;
+                    thereIsFood = false;
 
-                    addEat();
-                } else {
-                    isEndGame = true;
+                    addEatOrItem();
+                }
+                else if(getPawn(coordinates).getPawn().isItem()) {

```

Comment:
Sen sanırım senin eklediğin item a genel olarak kodda item demişsin. Fakat senin eklediğin yalnızca bir item çeşidi. O yüzden kodda kendi eklediğin item ile ilgili spesifik isimlendirmeler kullan.

Reply:
Done